src/main.ts

```diff
@@ -215,4 +219,38 @@ function dependabotJobUrl(id: number): string {
   return url_parts.filter(Boolean).join('/')
 }
 
+function credentialsFromEnv(): Credential[] {
+  const registriesProxyStr = process.env.GITHUB_REGISTRIES_PROXY
+  let credentialsStr: string
+  if (registriesProxyStr !== undefined) {
+    credentialsStr = Buffer.from(registriesProxyStr, 'base64').toString()
+  } else {
+    return []
+  }
+
+  let parsed: Credential[]
+
+  try {
+    parsed = JSON.parse(credentialsStr) as Credential[]
+  } catch {
+    // Don't log the error as it may contain sensitive information
+    parsed = []
+    botSay('Failed to parse GITHUB_REGISTRIES_PROXY environment variable')
+  }
+
+  for (const e of parsed) {
+    // Mask credentials to reduce chance of accidental leakage in logs.
+    if (e.password !== undefined) {
+      core.setSecret(e.password)
+    }
+    if (e.token !== undefined) {
+      core.setSecret(e.token)
```

Comment:
`auth-key` is another secret that could be masked. I do wish we had a `is-secret` flag on these.

Reply:
Yup. I wonder if we make it future proof by setting all properties as secret.

Reply:
Suggested future-proof version below.